docs/05_best-practices/04_data-design-and-migration.md

```diff
@@ -2,9 +2,9 @@
 content_title: Data design and migration
 ---
 
-EOSIO based blockchains allow developers to easily update their smart contract code. However, a few things need to be considered when it comes to data update and/or migration. The main structure for storing data in EOSIO based blockchains is the multi index table. Once a multi index table has been created with a first version of a smart contract, it has some limitations when it comes to changing its structure. Below you will find a few possible approaches which you can consider when you design your smart contract data and its migration. 
+EOSIO based blockchains allow developers to easily update their smart contract code. However, a few things need to be considered when it comes to data update and/or migration. The main structure for storing data in EOSIO based blockchains is the multi-index table. Once a multi-index table has been created with a first version of a smart contract, it has some limitations when it comes to changing its structure. Below you will find a few possible approaches which you can consider when you design your smart contract data and its migration. 
```

Comment:
However, a few things need to be considered when it comes to data updateS and/or migration.

Reply:
RE: The main structure for storing data in EOSIO based blockchains is the multi-index table. 
Would it be better to talk about multi-index table as a data store API? This matches the KV-API.

The main structure for storing data in EOSIO based blockchains is the multi-index table => becomes something like:
The multi-index table APi is one of the mechanisms (KV-API being the other) for storing and updating blockchain state. The multi-index table APi creates and uses data structures in RAM,. Changing these structures, i.e. adding new fields ... etc

Reply:
If KV-API is an API, Multi-Index Tables is also an API => the data store is chainbase or rocksdb (although chainbase uses boost multi index tables and memory mapped files under the covers.)